proposals/p1927.md

```diff
@@ -0,0 +1,96 @@
+# Slice generator
+
+<!--
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+[Pull request](https://github.com/carbon-language/carbon-lang/pull/1927)
+
+<!-- toc -->
+
+## Table of contents
+
+-   [Problem](#problem)
+-   [Background](#background)
+-   [Proposal](#proposal)
+-   [Details](#details)
+-   [Rationale](#rationale)
+-   [Alternatives considered](#alternatives-considered)
+
+<!-- tocstop -->
+
+## Problem
+
+Slices of arrays is a popular way to extract subarray of array. They also may
+be used for generating arrays.
+
+## Background
+
+There is no requirements for background.
+
+## Proposal
+
+Introduce slice generators.
+
+## Details
+
+Slice syntax is `a:b:s`. Here, `a` is the beginning of slice, `b` is the end of
+slice and `s` is the step in slice. In array indexing, `:s` can be omitted, then
+it means that step is 1. If `a` is omitted, slice starts from the beginning of
+array. The same situation is for `b`, but slice ends at the end of arrays. `:`
+is the same as `::`. If `s` is negative, lower and upper bounds are replaces
+each other.
+
+Slice generator may return `i32`, `i64`, `i128` and unsigned versions of them.
+
+Slice as generator of array:
+```carbon
+  var array: auto = (1:5:2); // array = (1,3,5)
+  var arrav: auto = 1:5:2; // arrav = (1,3,5)
+```
+Round brackets may be omitted.
+
+In operator `[]` of array, slices can be used in the following ways:
+```
+  var array: auto = (1:5:2); // array = (1,3,5)
+  var a1: auto = array[:]; // a1 = (1,3,5)
+  var a2: auto = array[::]; // a2 = (1,3,5)
+  var a3: auto = array[:1]; // a3 = (1,3)
+  var a4: auto = array[::-1]; // a4 = (5,3,1)
+  var a5: auto = array[:2:-1]; // a5 = (5)
+```
+
+Note, `auto` for slice of one element gives an array containing one element.
+
+## Rationale
```

Comment:
Similar to [the feedback being given on #1928](https://github.com/carbon-language/carbon-lang/pull/1928/files#r940721704), I recommend going through other accepted, merged proposals to see what the expectations are for writing rationale and alternatives. Specifically, the rationale needs explain _why_ the proposed approach is good to choose based on the goals, and the alternatives should compare advantages and disadvantages of the options.

Reply:
Hope, it is more clear now.